File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/offline/OfflineRegionStatus.java
package org.maplibre.nativeffi.offline;

/** Offline region status snapshot copied from native event payloads. */
public record OfflineRegionStatus(
    OfflineRegionDownloadState downloadState,
    int rawDownloadState,
    long completedResourceCount,
    long completedResourceSize,
    long completedTileCount,
    long requiredTileCount,
    long completedTileSize,
    long requiredResourceCount,
    boolean requiredResourceCountIsPrecise,
    boolean complete) {}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/offline/OfflineRegionStatus.kt
package org.maplibre.nativeffi.offline

/** Offline region status snapshot copied from native storage. */
public data class OfflineRegionStatus(
  public val downloadState: OfflineRegionDownloadState,
  public val rawDownloadState: Int,
  public val completedResourceCount: Long,
  public val completedResourceSize: Long,
  public val completedTileCount: Long,
  public val requiredTileCount: Long,
  public val completedTileSize: Long,
  public val requiredResourceCount: Long,
  public val requiredResourceCountIsPrecise: Boolean,
  public val complete: Boolean,
)
